includes/class-wc-calypso-bridge-setup.php

```diff
@@ -86,6 +102,38 @@ public function prevent_redirects_on_activation( $location, $status ) {
 		return $location;
 	}
 
+	/**
+	 * Handle the store location's onboarding redirect when user provided a full address.
+	 *
+	 * @since 1.9.4
+	 * @return void
+	 */
+	public function redirect_store_details_onboarding() {
+
+		// Only run on save.
+		if ( empty( $_POST ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing
+			return;
+		}
+
+		if ( ! isset( $_GET['tutorial'], $_GET['page'] ) || 'true' !== $_GET['tutorial'] || 'wc-settings' !== $_GET['page'] ) {
+			return;
+		}
+
+		$should_redirect_home = false;
+
+		$store_address  = get_option( 'woocommerce_store_address' );
+		$store_city     = get_option( 'woocommerce_store_city' );
+		$store_postcode = get_option( 'woocommerce_store_postcode' );
+
+		if ( ! empty( $store_address ) && ! empty( $store_city ) && ! empty( $store_postcode ) ) {
+			$should_redirect = true;
+		}
+
+		if ( $should_redirect ) {
+			wp_safe_redirect( admin_url( 'admin.php?page=wc-admin' ) );
+		}
```

Comment:
There seems to be a mismatch between `$should_redirect_home` and `$should_redirect`? Is that needed at all?
```suggestion
		$store_address  = get_option( 'woocommerce_store_address' );
		$store_city     = get_option( 'woocommerce_store_city' );
		$store_postcode = get_option( 'woocommerce_store_postcode' );

		if ( ! empty( $store_address ) && ! empty( $store_city ) && ! empty( $store_postcode ) ) {
			wp_safe_redirect( admin_url( 'admin.php?page=wc-admin' ) );
		}
```

Reply:
Nice catch! 

> Is that needed at all?

We can definitely work without it ;) 

Cleaned up in https://github.com/Automattic/wc-calypso-bridge/pull/811/commits/93fab1a089c5f27343c713dac97c6f1dedcff966